routes/router.go

```diff
@@ -55,6 +56,12 @@ func (r *Router) loadRoutes() {
 
 	r.router.Use(middleware2.BodyLimit("15M"))
 
+	r.router.HTTPErrorHandler = r.views.CustomHTTPErrorHandler
+
+	r.router.Use(middleware2.GzipWithConfig(middleware2.GzipConfig{
+		Level: 5,
+	}))
```

Comment:
Is this necessary if we have nginx in front anyway?

Reply:
So I've done some digging, and as far as I can tell in the default configuration (which I believe we use) nginx won't send an `Accept-Encoding` header to the upstream, which means this won't do anything. To benefit from this we'd need to configure nginx to forward the client's `Accept-Encoding` (and there may also be issues with nginx using HTTP/1.0 for proxied requests whereas this needs 1.1).

So in theory this may be beneficial (as we transfer less data in between nginx and the backend - we'd need to pay the CPU cost of gzipping at some point), but we need to double-check our nginx configs otherwise this may be redundant.

Reply:
I did that and checked the size of transfer and it can be upwards of MBs especially for all users

Reply:
I can have it as an env to be disabled?

Reply:
Nah, I think it makes sense to enable it (it does nothing without the `Accept-Encoding` set anyway), we just need to double-check that our nginx is set up properly to benefit from it, otherwise it'll do nothing

Reply:
Okay that's fine, I can do some googling later